app/js/app/directives/react_classes/RegexPatternBlock.js

```diff
@@ -0,0 +1,99 @@
+define(["react", "jquery"], function(React,$) {
+	return function(_ContentEditor, Block, ContentBlock, ContentValueOrChildren) {
+		return React.createClass({
+
+			onDocChange: function(_c, oldDoc, newDoc) {
+				this.props.onChange(this, oldDoc, newDoc);
+			},
+
+			onValueChange: function(e) {
+				var oldDoc = this.props.doc;
+				var newDoc = $.extend({}, oldDoc);
+				newDoc.value = e.target.value;
+				this.onDocChange(this, oldDoc, newDoc);
+			},
+
+			onExplanationChange: function(_c, _oldVal, newVal) {
+				var oldDoc = this.props.doc;
+				var newDoc = $.extend({}, oldDoc);
+				newDoc.explanation = newVal;
+				this.onDocChange(this, oldDoc, newDoc);
+			},
+
+			matchWholeString_toggle: function(_e) {
+				var oldDoc = this.props.doc;
+				var newDoc = $.extend({}, oldDoc);
+				newDoc.matchWholeString = !oldDoc.matchWholeString;
+				this.onDocChange(this, oldDoc, newDoc);
+			},
+
+			caseInsensitive_toggle: function(_e) {
+				var oldDoc = this.props.doc;
+				var newDoc = $.extend({}, oldDoc);
+				newDoc.caseInsensitive = !oldDoc.caseInsensitive;
+				this.onDocChange(this, oldDoc, newDoc);
+			},
+
+			multiLineRegex_toggle: function(_e) {
+				var oldDoc = this.props.doc;
+				var newDoc = $.extend({}, oldDoc);
+				newDoc.multiLineRegex = !oldDoc.multiLineRegex;
+				this.onDocChange(this, oldDoc, newDoc);
+			},
+
+			regexHelper: function(_e) {
+				var regex = this.props.doc.value
+				if (this.props.doc.matchWholeString) {
+					// Add caret and dollar sign if the whole string should be matched
+					regex = "^" + regex + "$"
+				}
+				var flags = "g" + (this.props.doc.multiLineRegex ? "m" : "") + (this.props.doc.caseInsensitive ? "i" : "")
+				window.open(`https://regex101.com/?regex=${encodeURI(regex)}&flags=${flags}&delimiter=@&flavor=java`)
```

Comment:
It looks (from regex101/ the example 'correct' answer on your demo page) like this should be javascript?

Reply:
Do you mean that the 'flavor' should be javascript? Otherwise I'm not sure sorry :/

Reply:
Forgive me my drive-by comment, but is `encodeURI` the right function to use, or do you want `encodeURIComponent`? I have been bitten by this before, and I have a feeling you want the latter for when regexes contain question marks!

Reply:
I've just checked both out and I think you're right - `encodeURIComponent` is [apparently](https://stackoverflow.com/questions/75980/when-are-you-supposed-to-use-escape-instead-of-encodeuri-encodeuricomponent) used if you want to encode a query parameter which is exactly what I'm doing here. it's a quick change, I'll push it in a second.

Reply:
As it turns out, the test I was using already contained question marks and worked fine with `encodeURI`, but as per the SO answer and the function names `encodeURIComponent` is probably most suitable.

Reply:
Now that I think about it, most URL parsers will ignore repeated question marks. Ampersands are your real headache for a URL param (since that ends the param), though they don't occur often in regexes at least. Quick fix though!

Reply:
Also, I agree that the flavour should be Java, so I don't know what @bh412's comment is about and I'm closing this!